lib/writeGltf.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+var Cesium = require('cesium');
+var mime = require('mime');
+var PNG = require('pngjs').PNG;
+var Promise = require('bluebird');
+var getBufferPadded = require('./getBufferPadded');
+var gltfToGlb = require('./gltfToGlb');
+
+var defined = Cesium.defined;
+var RuntimeError = Cesium.RuntimeError;
+
+module.exports = writeGltf;
+
+/**
+ * Write glTF resources as embedded data uris or external files.
+ *
+ * @param {Object} gltf The glTF asset.
+ * @param {Object} options The options object passed along from lib/obj2gltf.js
+ * @returns {Promise} A promise that resolves to the glTF JSON or glb buffer.
+ *
+ * @private
+ */
+function writeGltf(gltf, options) {
+    return encodeTextures(gltf)
+        .then(function() {
+            var binary = options.binary;
+            var separate = options.separate;
+            var separateTextures = options.separateTextures;
+
+            var promises = [];
+            if (separateTextures) {
```

Comment:
So it's a bit hard to notice, but if `separate` is true `separateTextures` will also be true.

In `obj2gltf.js`
```
options.separateTextures = defaultValue(options.separateTextures, defaults.separateTextures) || options.separate;
```